RTL87xx Compile error with 2025.7.0

### The problem


### Which version of ESPHome has the issue?

2025.7.0

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

RTL87XX

### Component causing the issue

_No response_

### YAML Config

```yaml
substitutions:
  hostname: "sirene"
  devicename: "sirene"
  friendly_name: "Sirene"
  device_description: "LSC Sirene"
  ip_address: 192.168.x,x


esphome:
  name: ${hostname}
  friendly_name: ${friendly_name}

rtl87xx:  # RTL8710BN
  board: wr3

logger:

api:
  encryption:
    key: !secret api_encryption_key

#web_server:
#  port: 80

ota:
  - platform: esphome
    password: !secret ota_password
#  - platform: web_server

wifi:
  ssid: !secret wifi_iot_ssid
  password: !secret wifi_iot_password
  fast_connect: on
  use_address: ${ip_address}
  ap:  
    ssid: !secret wifi_ap_ssid
    password: !secret wifi_ap_password

switch:
  - platform: restart
    name: Restart
  - platform: gpio
    pin: PA05
    name: Licht
    id: licht

binary_sensor:
  - platform: gpio
    name: PairOnOff
    pin:
      number: PA15
      inverted: true
      mode:
        input: true

sensor:
    - platform: wifi_signal
      name: Wifi

output:
  - platform: libretiny_pwm
    pin: PA12
    frequency: 2600 Hz
    id: pwm_output

text_sensor:
  - platform: libretiny
    version:
      name: LibreTiny Version
light:
  - platform: monochromatic
    output: pwm_output
    name: Sirene
    id: sirene

button:
- platform: template
  name: Buzzer
  on_press:
  - script.execute: buzzer_five

script: 
- id: buzzer_five
  then:
    - repeat:
        count: 10
        then:
          #- light.turn_on: sirene
          - switch.turn_on: licht
          - delay: 500ms                
          #- light.turn_off: sirene
          - switch.turn_off: licht
          - delay: 1000ms
```

### Anything in the logs that might be useful for us?

```txt
INFO ESPHome 2025.7.0
INFO Reading configuration /config/esphome/sirene.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing sirene (board: wr3; framework: arduino; platform: libretiny @ 1.9.1)
--------------------------------------------------------------------------------
Warning! Non-Git installations are NOT SUPPORTED.
HARDWARE: RTL8710BN 125MHz, 256KB RAM, 788KB Flash
 - framework-arduino-api @ 2022.8.24+sha.237b10a 
 - framework-realtek-amb1 @ 0.0.0+v2022.06.21.sha.c4e44ef 
 - library-flashdb @ 1.2.0+sha.d5c892f 
 - library-freertos @ 8.1.2+sha.776ae6c 
 - library-freertos-port @ 2025.1.18+sha.dbfc408 
 - library-lwip @ 2.1.3-amb1+sha.6297b80 
 - library-printf @ 6.1.0+sha.28a79bd 
 - library-uf2ota @ 5.0.0+sha.f955412 
PLATFORM VERSIONS:
 - libretiny @ 1.9.1
 - ltchiptool @ 4.12.0
CUSTOM OPTIONS:
 - fw_name = esphome
 - fw_version = 2025.7.0
Dependency Graph
|-- noise-c @ 0.1.10
Compiling .pioenvs/sirene/src/main.cpp.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_core/salsa/ref/core_salsa_ref.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_hash/crypto_hash.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_hash/sha256/cp/hash_sha256_cp.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_hash/sha256/hash_sha256.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_onetimeauth/crypto_onetimeauth.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_onetimeauth/poly1305/donna/poly1305_donna.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_onetimeauth/poly1305/onetimeauth_poly1305.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_scalarmult/crypto_scalarmult.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_scalarmult/curve25519/ref10/x25519_ref10.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_scalarmult/curve25519/scalarmult_curve25519.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_stream/chacha20/ref/chacha20_ref.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_stream/chacha20/stream_chacha20.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_stream/crypto_stream.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_stream/salsa20/ref/salsa20_ref.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_stream/salsa20/stream_salsa20.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_stream/salsa20/xmm6/salsa20_xmm6-asm.S.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_stream/salsa20/xmm6/salsa20_xmm6.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_stream/salsa20/xmm6int/salsa20_xmm6int-avx2.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_stream/salsa20/xmm6int/salsa20_xmm6int-sse2.c.o
Compiling .pioenvs/sirene/lib510/libsodium/crypto_verify/verify.c.o
In file included from src/esphome/core/event_pool.h:8,
                 from src/esphome.h:95,
                 from src/main.cpp:3:
src/esphome/core/lock_free_queue.h: In member function 'bool esphome::NotifyingLockFreeQueue<T, SIZE>::push(T*)':
src/esphome/core/lock_free_queue.h:133:7: error: there are no arguments to 'xTaskNotifyGive' that depend on a template parameter, so a declaration of 'xTaskNotifyGive' must be available [-fpermissive]
  133 |       xTaskNotifyGive(task_to_notify_);
      |       ^~~~~~~~~~~~~~~
src/esphome/core/lock_free_queue.h:133:7: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
Compiling .pioenvs/sirene/lib510/libsodium/randombytes/internal/randombytes_internal_random.c.o
Compiling .pioenvs/sirene/lib510/libsodium/randombytes/randombytes.c.o
Compiling .pioenvs/sirene/lib510/libsodium/randombytes/sysrandom/randombytes_sysrandom.c.o
```

### Additional information

With the previous ESPHome 2025.6.3 it compiled without errors

--- a/esphome/core/event_pool.h
+++ b/esphome/core/event_pool.h
@@ -1,6 +1,6 @@
 #pragma once
 
-#if defined(USE_ESP32) || defined(USE_LIBRETINY)
+#if defined(USE_ESP32)
 
 #include <atomic>
 #include <cstddef>
@@ -78,4 +78,4 @@ template<class T, uint8_t SIZE> class EventPool {
 
 }  // namespace esphome
 
-#endif  // defined(USE_ESP32) || defined(USE_LIBRETINY)
+#endif  // defined(USE_ESP32)

--- a/esphome/core/lock_free_queue.h
+++ b/esphome/core/lock_free_queue.h
@@ -1,17 +1,12 @@
 #pragma once
 
-#if defined(USE_ESP32) || defined(USE_LIBRETINY)
+#if defined(USE_ESP32)
 
 #include <atomic>
 #include <cstddef>
 
-#if defined(USE_ESP32)
 #include <freertos/FreeRTOS.h>
 #include <freertos/task.h>
-#elif defined(USE_LIBRETINY)
-#include <FreeRTOS.h>
-#include <task.h>
-#endif
 
 /*
  * Lock-free queue for single-producer single-consumer scenarios.
@@ -148,4 +143,4 @@ template<class T, uint8_t SIZE> class NotifyingLockFreeQueue : public LockFreeQu
 
 }  // namespace esphome
 
-#endif  // defined(USE_ESP32) || defined(USE_LIBRETINY)
+#endif  // defined(USE_ESP32)

